tokio/src/runtime/coop.rs

```diff
@@ -240,6 +243,42 @@ cfg_coop! {
             self.0.is_none()
         }
     }
+
+    pin_project! {
+        /// A future type that calls `poll_proceed` before polling the inner future to check if the
+        /// inner future has exceeded its budget. If the inner future resolves, this will
+        /// automatically call `RestoreOnPending::made_progress` before resolving this future with
+        /// the result of the inner one. If polling the inner future is pending, polling this future
+        /// type will also return a `Poll::Pending`.
+        #[must_use = "futures do nothing unless polled"]
+        pub(crate) struct BudgetConstraintFuture<F: Future> {
```

Comment:
This is a pretty long name. We could just call it `Coop` or something like that. Thoughts?

Reply:
`Coop` sounds pretty good and simple. Naming is hard.
I wasn't sure what the naming convention here is for those simple future types as there are some types with `Future` at the end of the type name and some without.

Reply:
I don't think we have a hard naming convention for them.